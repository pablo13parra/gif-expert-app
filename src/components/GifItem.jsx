
import PropTypes from 'prop-types';
const GifItem = ({ title, url, id }) => {
    return (
        <div className="card">
            <h4>{title}</h4>
            <img src={url} alt="altxd" />
        </div>
    )
}

export default GifItem;
GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    id: PropTypes.string
}