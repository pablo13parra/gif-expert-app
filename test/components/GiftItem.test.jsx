import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('GiftItem test', () => {
    const image = {
        title: 'Saitama',
        url: 'https://one-punch.com/saitama.jpg',
        id: '0'
    }
    test('should match with snapshot', () => {
        const { container } = render(<GifItem key={image.id} {...image} />);
        expect(container).toMatchSnapshot();
    });

    test('should show the alt and url correctly', () => {
        render(<GifItem key={image.id} {...image} />);
        const { src, alt } = screen.getByRole('img');
        //ver que es lo que estoy renderizando
        // screen.debug();
        expect(src).toBe(image.url);
        expect(alt).toBe('altxd');
    });

    test('should show the title on the component', () => { 
        render(<GifItem key={image.id} {...image} />);
        const cardTitle = screen.getByRole('heading', {level:4})
        const content = cardTitle.textContent;
        expect(content).toBeTruthy();
        expect(content).toContain(image.title)
        // expect(screen.getByRole('h4'))
     })

})