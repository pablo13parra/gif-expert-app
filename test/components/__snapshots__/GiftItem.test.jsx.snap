// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItem test should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <h4>
      Saitama
    </h4>
    <img
      alt="altxd"
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
